src/runtime/math.go

```diff
@@ -169,28 +178,59 @@ func math_log2(x float64) float64
 
 //go:linkname math_Max math.Max
 func math_Max(x, y float64) float64 {
-	if GOARCH == "arm64" || GOARCH == "wasm" {
-		return llvm_maximum(x, y)
+	if GOARCH == "arm64" || GOARCH == "wasm" || GOARCH == "i386" || GOARCH == "amd64" {
+		if math.IsInf(x, 1) || math.IsInf(y, 1) {
+			return math.Inf(1)
+		}
+		if math.IsNaN(x) || math.IsNaN(y) {
+			return math.NaN()
+		}
+		// test for zero regardless of sign
+		if y == 0 && x == 0 {
+			// if both are negative zero
+			if math.Signbit(x) && math.Signbit(y) {
+				return x // returns -0.0
+			}
+			return 0
+		}
+		return llvm_maxnum(x, y)
 	}
 	return math_max(x, y)
 }
 
-//export llvm.maximum.f64
-func llvm_maximum(x, y float64) float64
+//export llvm.maxnum.f64
+func llvm_maxnum(x, y float64) float64
 
 //go:linkname math_max math.max
 func math_max(x, y float64) float64
 
 //go:linkname math_Min math.Min
 func math_Min(x, y float64) float64 {
-	if GOARCH == "arm64" || GOARCH == "wasm" {
-		return llvm_minimum(x, y)
+	if GOARCH == "arm64" || GOARCH == "wasm" || GOARCH == "i386" || GOARCH == "amd64" {
+		if math.IsInf(x, -1) || math.IsInf(y, -1) {
+			return math.Inf(-1)
+		}
+		if math.IsNaN(x) || math.IsNaN(y) {
+			return math.NaN()
+		}
+		// test for zero regardless of sign
+		if y == 0 && x == 0 {
+			// if both are negative zero
+			if math.Signbit(x) {
+				return x // returns -0.0
+			}
+			if math.Signbit(y) {
+				return y // returns -0.0
+			}
+			return 0
+		}
```

Comment:
With all of this complexity (and with the requirement to import the math package from the runtime!) I would actually much prefer to just use the standard Go  versions. This is basically a re-implementation of the version in the math package and just as long.

Reply:
Fully agree.